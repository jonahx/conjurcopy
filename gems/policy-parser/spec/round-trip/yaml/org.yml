# An example of an organization built with Conjur DSL.
# In practice this would be loaded in multiple steps:
#
# Step 1 - conjur bootstrap - global data
# Step 2 - Users and groups imported from LDAP
# Step 3 - Load policies
# Step 4 - Load entitlements
# 
# This document shows Step 1 and Step 2

# Step 1 - Create the global group "everyone"

# Global group fixture.
- !group
  id: everyone

# Step 2 - Load users and groups
  
# Groups imported from LDAP
- !group
  id: role_messaging
  gidnumber: 10403
  annotations:
    "ldap-sync/source": "ldap://util4002.ldap.internal"
- !group
  id: role_metadata
  gidnumber: 10404
  annotations:
    "ldap-sync/source": "ldap://util4002.ldap.internal"
- !group
  id: role_storage
  gidnumber: 10405
  annotations:
    "ldap-sync/source": "ldap://util4002.ldap.internal"

# Users imported from LDAP
- !user
  id: alice
  uidnumber: 1003
  annotations:
    "ldap-sync/source": "ldap://util4002.ldap.internal"
- !user
  id: bob
  uidnumber: 1004
  annotations:
    "ldap-sync/source": "ldap://util4002.ldap.internal"
- !user
  id: charles
  uidnumber: 1005
  annotations:
    "ldap-sync/source": "ldap://util4002.ldap.internal"

# Ensure that all the LDAP users and groups are included in "everyone".
- !grant
  role: !group everyone
  members:
  - !group role_messaging
  - !group role_metadata
  - !group role_storage
  - !user alice
  - !user bob
  - !user charles

# Place the users into groups.
- !grant
  role: !group role_messaging
  member: !user alice
- !grant
  role: !group role_metadata
  member: !user bob
  
# Permit the "everyone" group to read every user
- !permit
  privilege: read
  role: !group everyone
  resources: 
  - !user alice
  - !user bob
  - !user charles
