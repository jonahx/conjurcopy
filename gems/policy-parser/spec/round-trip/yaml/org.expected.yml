---
- !group
  id: everyone
- !group
  annotations:
    ldap-sync/source: ldap://util4002.ldap.internal
  gidnumber: 10403
  id: role_messaging
- !group
  annotations:
    ldap-sync/source: ldap://util4002.ldap.internal
  gidnumber: 10404
  id: role_metadata
- !group
  annotations:
    ldap-sync/source: ldap://util4002.ldap.internal
  gidnumber: 10405
  id: role_storage
- !user
  annotations:
    ldap-sync/source: ldap://util4002.ldap.internal
  id: alice
  uidnumber: 1003
- !user
  annotations:
    ldap-sync/source: ldap://util4002.ldap.internal
  id: bob
  uidnumber: 1004
- !user
  annotations:
    ldap-sync/source: ldap://util4002.ldap.internal
  id: charles
  uidnumber: 1005
- !grant
  member:
  - !member
    role: !group
      id: role_messaging
  - !member
    role: !group
      id: role_metadata
  - !member
    role: !group
      id: role_storage
  - !member
    role: !user
      id: alice
  - !member
    role: !user
      id: bob
  - !member
    role: !user
      id: charles
  role: !group
    id: everyone
- !grant
  member: !member
    role: !user
      id: alice
  role: !group
    id: role_messaging
- !grant
  member: !member
    role: !user
      id: bob
  role: !group
    id: role_metadata
- !permit
  privilege: read
  resource:
  - !user
    id: alice
  - !user
    id: bob
  - !user
    id: charles
  role: !group
    id: everyone
